base/src/norm.cu

```diff
@@ -134,53 +138,63 @@ class Norm_1x1< Vector<TemplateConfig<t_memSpace, t_vecPrec, t_matPrec, t_indPre
             double sum = 0.l;
             A.getOffsetAndSizeForView(OWNED, &offset, &size);
 
-            switch (norm_type)
+            if (norm_type == L1 || norm_type == L1_SCALED)
             {
-                case L1:
-                    block_nrm[0] = nrm1(r, offset, size);
-
-                    if (A.is_matrix_distributed())
-                    {
-                        A.getManager()->getComms()->global_reduce(values, block_nrm, A, 4);
-                        block_nrm[0] = 0;
+                block_nrm[0] = nrm1(r, offset, size);
 
-                        for (int j = 0; j < values.size(); j++) { sum += values[j][0]; }
+                if (A.is_matrix_distributed())
+                {
+                    A.getManager()->getComms()->global_reduce(values, block_nrm, A, 4);
+                    block_nrm[0] = 0;
 
-                        block_nrm[0] = sum;
+                    for (int j = 0; j < values.size(); j++)
+                    {
+                        sum += values[j][0];
                     }
 
-                    break;
-
-                case L2:
-                    block_nrm[0] = nrm2(r, offset, size);
+                    block_nrm[0] = sum;
+                }
 
-                    if (A.is_matrix_distributed())
-                    {
-                        block_nrm[0] *= block_nrm[0];
-                        A.getManager()->getComms()->global_reduce(values, block_nrm, A, 5);
-                        block_nrm[0] = 0;
+                if (norm_type == L1_SCALED)
+                {
+                    block_nrm[0] /= norm_factor;
```

Comment:
Just to double check - `norm_factor` is only used and computed in `L1_SCALED`, right?